### PR TITLE
Adding the SNS Client Document

### PR DESCRIPTION
**Pull request by VenkyOS**
Signed-off-by: Venkataraman Padmanabhan <venkataraman.padmanabhan@seagate.com>

### CODE REVIEW

File: doc/HLD_of_SNS_Client.rst [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+  - [u.LAYOUT.DATA] ST: layouts for data are supported
+
+  - [u.LAYOUT.POOLS] ST: layouts use server and device pools for object allocation, location, and identification
+
+  - [u.LAYOUT.PARAMETRIZED] ST: layouts have parameters that are substituted to perform actual mapping
+
+- fop:
+
+  - [u.fop] ST: M0 uses File Operation Packets (FOPs)
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Client.rst:141]: Please use upper case letter or lower case letter in all places. Don't mix them.

File: doc/HLD_of_SNS_Client.rst [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+Environment: normal client operation
+
+Artifact: call to VFS ->read() entry point
+
+Response: a fop is created, network transmission of operation parameters to all involved data servers is started as specified by the file layout, servers place retrieved data directly in user buffers, once transmission completes, the fop is destroyed.
+
+Response measure: no data copying in the process
+
+Questions and issues:
+
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Client.rst:213]: Can you please convert this into a table?
Please refer to the previous PR I have done.

File: doc/HLD_of_SNS_Client.rst [hunk 3]
```diff
@@ -0,0 +1,283 @@
+==================================================
+High level design of an SNS client module for M0
+==================================================
+
+This document presents a high level design (HLD) of an SNS client module from M0. The main purposes of this document are: (i) to be inspected by M0 architects and peer designers to ascertain that high level design is aligned with M0 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document.
```
[the remark this answers is not in the document]
Comment (VenkyOS) [reply doc/HLD_of_SNS_Client.rst]: Incorporated.

File: doc/HLD_of_SNS_Client.rst [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+- network transfer is a process of transmitting operation code and operation parameters, potentially including data pages, to the corresponding data server, and waiting for server reply. Network transfer is implemented as a synchronous call to M0 rpc service;
+
+- an IO operation consists of (IO) updates, each accessing or modifying file system state on a single server. Updates of the same IO operation are siblings.
+
+*************
+Requirements
+*************
+
+- [r.sns-client.nocache]: Motr client does not cache data because (i) this simplifies its structure and implementation, (ii) Motr exports block device interface and block device users, including file systems, do caching internally;
```
[the remark this answers is not in the document]
Comment (VenkyOS) [reply doc/HLD_of_SNS_Client.rst]: Incorporated.

File: doc/HLD_of_SNS_Client.rst [hunk 5]
```diff
@@ -0,0 +1,283 @@
+==================================================
+High level design of an SNS client module for M0
+==================================================
+
+This document presents a high level design (HLD) of an SNS client module from M0. The main purposes of this document are: (i) to be inspected by M0 architects and peer designers to ascertain that high level design is aligned with M0 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document.
+
+The intended audience of this document consists of M0 customers, architects, designers and developers.
```
[the remark this answers is not in the document]
Comment (VenkyOS) [reply doc/HLD_of_SNS_Client.rst]: Deferred.

File: doc/HLD_of_SNS_Client.rst [hunk 6]
```diff
@@ -0,0 +1,283 @@
+==================================================
+High level design of an SNS client module for M0
```
[the remark this answers is not in the document]
Comment (VenkyOS) [reply doc/HLD_of_SNS_Client.rst]: Incorporated.

File: doc/HLD_of_SNS_Client.rst (hunk 1 above)
Comment (VenkyOS) [reply doc/HLD_of_SNS_Client.rst]: Incorporated.

File: doc/HLD_of_SNS_Client.rst (hunk 2 above)
Comment (VenkyOS) [reply doc/HLD_of_SNS_Client.rst]: Incorporated.

**Pull request merged**